File: kinny-sellergoods-interface/src/main/java/com/kinny/sellergoods/service/BrandService.java
package com.kinny.sellergoods.service;

import com.kinny.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * @author qgy
 * @create 2019/5/2 - 16:05
 */
public interface BrandService {

    /**
     *  查询全部品牌数据
     * @return
     */
    public List<TbBrand> findAll();

    /**
     *  根据当前页数 以及 一页条数分页查询 使用map 易于扩展参数 不用修改接口
     * @param map
     * @return
     */
    public Map<String, Object> findPage(Map<String, Object> map);

    /**
     *  根据id查询品牌数据
     * @param id
     * @return
     */
    public TbBrand findOne(String id);


    /**
     *  保存品牌 新增或者修改 根据是否含有记录唯一标识
     * @param brand
     */
    public void save(TbBrand brand);

    /**
     *  根据主键集合批量删除品牌
     * @param ids
     */
    public void batchDelete(List<String> ids);

}

File: kinny-manager-web/src/main/webapp/js/service/BrandService.js
// 创建服务 分层开发 易于维护的扩展 增强可读性
/**
 *  $http angularjs 内置服务 异步访问后台api接口 可以作为实参传递过来 依赖注入
 *  主键按照一种预先定义好的方式接受容器的资源注入
 */
app.service('brandService', function ($http) {

    // js 对象添加属性 函数 this引用的是回调函数所在对象
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };

    this.findAsPage = function (pageEntity) {
       return $http.get('../brand/findPage.do?pageIndex=' + pageEntity.currentPage + "&pageSize=" + pageEntity.itemsPerPage);
    }
    
    this.save = function (entity) {
        return $http.post('../brand/save.do', entity);
    }

    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    }

    this.delete = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    }
    

});